src/apps/user/routers.py

```diff
@@ -84,24 +80,47 @@ def get_user(user_id: int, db: Session = Depends(get_db)) -> UserOutputSchema:
     return db_user
 
 
-@router.patch(
+@user_router.get(
+    "/{user_id}/orders",
+    response_model=PagedResponseSchema[OrderOutputSchema],
+    status_code=status.HTTP_200_OK,
+)
+def get_user_orders(
+    user_id: int,
+    db: Session = Depends(get_db),
+    page_params: PageParams = Depends(),
+    request_user: User = Depends(authenticate_user),
+) -> PagedResponseSchema[OrderOutputSchema]:
+    check_object_permission(user_id, request_user)
+    db_orders = get_all_user_orders(db, user_id, page_params)
+    return db_orders
+
+
+@user_router.patch(
     "/{user_id}",
-    dependencies=[Depends(authenticate_user)],
     response_model=UserOutputSchema,
     status_code=status.HTTP_200_OK,
 )
 def update_user(
-    user_id: int, user: UserUpdateSchema, db: Session = Depends(get_db)
+    user_id: int,
+    user: UserUpdateSchema,
+    db: Session = Depends(get_db),
+    request_user: User = Depends(authenticate_user),
 ) -> UserOutputSchema:
+    check_object_permission(user_id, request_user)
     db_user = update_single_user(db, user, user_id)
     return db_user
 
 
-@router.delete(
+@user_router.delete(
     "/{user_id}",
-    dependencies=[Depends(authenticate_user)],
     status_code=status.HTTP_204_NO_CONTENT,
 )
-def delete_user(user_id: int, db: Session = Depends(get_db)) -> Response:
+def delete_user(
+    user_id: int,
+    db: Session = Depends(get_db),
+    request_user: User = Depends(authenticate_user),
+) -> Response:
+    check_permission(request_user)
```

Comment:
good option would be deactivate user and keep their data instead of deleting the whole object with data

Reply:
next pr